modules/acton/src/external/ActOn/ActOnIntegration.ts

```diff
@@ -0,0 +1,78 @@
+import type { ObjectSharedContext } from "@restatedev/restate-sdk";
+import * as restate from "@restatedev/restate-sdk";
+import type {
+	ExternalIntegrationState,
+	IExternalIntegration,
+} from "common/external";
+import { ActOnListAPI } from "../../../../common/src/external/acton";
+import type { ActOnResponse } from "../../../../common/src/external/acton/ActOnListAPI";
+import type { LeadState, WebLead } from "../../types";
+
+interface ActOnIntegrationState extends ExternalIntegrationState {
+	Data?: {
+		LeadId: string;
+	};
+}
+
+export class ActOnIntegration
+	implements IExternalIntegration<LeadState, ActOnIntegrationState>
+{
+	Name!: "ActOn";
+	defaultState(): ActOnIntegrationState {
+		return {
+			SyncStatus: "NOT SYNCED",
+		};
+	}
+	async create(
+		state: ActOnIntegrationState,
+		context: ObjectSharedContext<WebLead>,
+	): Promise<ActOnIntegrationState> {
+		context.console.log(`Starting 'create' for ActOn Integration`);
+		const leadState = await context.getAll();
+		const response = await context.run("Create ActOn Lead", async () => {
+			const listId = leadState?.Lead?.Lead?.listId!;
+			delete leadState?.Lead?.Lead?.listId;
+			const ActOnRes = await ActOnListAPI.CreateContactAPI(
+				listId,
+				leadState?.Lead?.Lead,
+			).catch(
+				(e) =>
+					({
+						status: "failure",
+						message: e,
+					}) as ActOnResponse,
+			);
+			return ActOnRes;
+		});
+		if (response?.status !== "success") {
+			return {
+				...state,
+				SyncStatus: "ERROR",
+				ErrorInfo: {
+					Message: response.message,
+					ErrorDate: new Date(await context.date.now()).toISOString(),
+				},
```

Comment:
Use a human readable message and return the error in the details object for better logging
```suggestion
				ErrorInfo: {
					Message: "Failed to create Acton Lead",
					Details: {
						Response: serializeError(response)
					},
					ErrorDate: new Date(await context.date.now()).toISOString(),
				},
```